src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -579,20 +679,27 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
 
 struct SyncedLogStoreBuffer {
     buffer: VecDeque<(u64, LogStoreBufferItem)>,
+    current_size: usize,
     max_size: usize,
     next_chunk_id: ChunkId,
     metrics: KvLogStoreMetrics,
+    flushed_count: usize,
```

Comment:
Seems that we didn't update the `flushed_count` properly when we flush a chunk to storage on barrier (when we mark the `flushed` as true for the `StreamChunk` variant).

Reply:
I intended `flushed_count` to be incremented, just for `FlushedItem`, because even if some chunk was flushed to storage on barrier, the chunk still remains in the buffer, and is not rewritten to a `FlushedItem`. So it can still be read directly from buffer.

Only `FlushedItem` will need to be read from storage.